packages/blade/src/components/Tour/_decisions/decisions.md

```diff
@@ -0,0 +1,766 @@
+  /**
+   * Footer content
+   */
+  footer?: ({ gotToNext, goToPrev, stopTour, activeStep, totalSteps }) => React.ReactNode;
+  /**
+   * Popover title
+   */
+  title?: string;
+  /**
+   * Leading content placed before the title
+   *
+   * Can be any blade icon or asset.
+   */
+  titleLeading?: React.ReactNode;
+  /**
+   * Placement of Popover
+   * @default "top"
+   */
+  placement?: UseFloatingOptions['placement'];
+}
+
+type TourProps = {
+  /**
+   * Array of steps to be rendered
+   *
+   * The order of the steps will be the order in which they are rendered depending on the `activeStep` prop
+   */
+  steps: Step[];
+  /**
+   * Whether the tour is visible or not
+   */
+  isOpen: boolean;
+  /**
+   * Callback when the tour is opened or closed
+   */
+  onOpenChange: ({ isOpen: boolean }) => void;
+  /**
+   * Callback which fires when the tour has reached the last step.
+   */
+  onFinish: () => void;
+  /**
+   * Callback when the active step changes
+   */
+  onStepChange?: (step: number) => void;
+  /**
+   * Active step to be rendered
+   */
+  activeStep: number;
+}
+```
+
+### `TourStep` API
+
+TourStep is an enhancer component which is used to wrap the element that needs to be highlighted with a specific unique identifier.
+
+This component is needed because:
+
+- in react-native there is no `id` prop or `getElementById` platform API.
+- We don't expose `id` for every blade component, currently only Box does.
+
+But to keep the API consistent between web & native, we will also expose this component for web.
+
+> Note: on web, this component doesn't attach an `id` to the DOM element, instead it uses `ref` to collect the DOM element and save it to the state inside the `Tour` component, See below [discussions](#react-native-specifics) to know more about this approach.
+
+```jsx
+type TourStepProps = {
+  /**
+   * Unique identifier/name for the tour step
+   *
+   * This should be the same as the `name` prop of the element inside the `steps` array of the `Tour` component
+   */
+  name: string
+  /**
+   * React children with a single child which exposes a ref
+   */
+  children: React.ReactNode;
+};
+```
+
+Note that, in order for TourStep to work properly it needs access to it's children ref, so the children should expose a ref.
+
+## Usage
+
+```jsx
+import { Tour } from '@razorpay/blade/components';
+import type { TourSteps } from '@razorpay/blade/components';
+
+const Footer = ({ gotToNext, goToPrev, stopTour, activeStep, totalStep }) => {
+  const isLast = activeStep === totalStep;
+  const isFirst = activeStep === 0;
+  return (
+    <Box>
+      <Text>
+        {activeStep} / {totalStep}
+      </Text>
+      {!isFirst && <Button onClick={gotToNext}>Prev</Button>}
+      {isLast ? <Button onClick={stopTour}>Done</Button> : <Button onClick={goToPrev}>Next</Button>}
+    </Box>
+  );
+};
+
+const steps: TourSteps = [
+  {
+    name: 'step-1',
+    title: 'Step 1',
+    content: ({ activeStep }) => (
+      <Box>
```

Comment:
Do we need a Box wrapper always? Might not be intuitive for consumers

Reply:
No you don't need any wrapper. I'll remove these unneeded boxes.